### PR TITLE
Allow to fail on primitives in constructor with `failOnNullForPrimitives=true`

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-serialization/issues/880

### CODE REVIEW

**Review by graemerocher — APPROVED** [review]

File: serde-api/src/main/java/io/micronaut/serde/config/DeserializationConfiguration.java [hunk 1]
```diff
@@ -43,4 +43,14 @@ public interface DeserializationConfiguration {
      */
     @Bindable(defaultValue = StringUtils.FALSE)
     boolean isStrictNullable();
+
+    /**
+     /**
```
Comment (yawkat) [inline serde-api/src/main/java/io/micronaut/serde/config/DeserializationConfiguration.java]:
```suggestion
```

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java [hunk 2]
```diff
@@ -803,6 +817,7 @@ public static final class DerProperty<B, P> {
                 .orElse(null);
             this.explicitlyRequired = annotationMetadata.booleanValue(SerdeConfig.class, SerdeConfig.REQUIRED)
                 .orElse(false);
+            this.explicitlyRequiredForConstructor = explicitlyRequired || argument.isPrimitive() && failOnNullForPrimitives;
```
Comment (yawkat) [inline serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java:820]:
```suggestion
            this.explicitlyRequiredForConstructor = explicitlyRequired || (argument.isPrimitive() && failOnNullForPrimitives);
```

**Review by yawkat — APPROVED** [review]

File: serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java (hunk 2 above)
Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java:820]: Why? Having trouble reading it?

Comment (yawkat) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java:820]: yes, no idea about the precedence rules for the logical operators

Comment (dstepanov) [reply serde-support/src/main/java/io/micronaut/serde/support/deserializers/DeserBean.java:820]: Just like in math * / AND has a priority :D

**Pull request merged**